expression/builtin_arithmetic.go

```diff
@@ -66,8 +66,6 @@ func (c *arithmeticPlusFunctionClass) setFlenDecimal4Int(retTp, a, b *types.Fiel
 // setFlenDecimal4Real is called to set proper `Flen` and `Decimal` of return
 // type according to the two input parameter's types.
 func (c *arithmeticPlusFunctionClass) setFlenDecimal4RealOrDecimal(retTp, a, b *types.FieldType, isReal bool) {
-	retTp.Decimal = int(math.Max(float64(a.Decimal), float64(b.Decimal)))
-	retTp.Flen = types.UnspecifiedLength
 	if a.Decimal != types.UnspecifiedLength && b.Decimal != types.UnspecifiedLength {
```

Comment:
if a.Decimal **==** types.UnspecifiedLength **||** b.Decimal **==** types.UnspecifiedLength

we set flen to unspecified?